### PR TITLE
Adds the circle game

### PR DESCRIPTION
**Pull request by PKPenguin321**
Started by using the "circle" emote.

See: https://tgstation13.org/phpBB/viewtopic.php?f=9&t=11951

Sprite:
circled hand inhand

**Not featured in this PR** but featured in that thread: On-mob sprites (couldn't think of an easy way to do them), starting the game by "using" an empty hand (it's started with an emote instead), messages telling you who examined you.

🆑
add: You can now play the circle game and prank your coworkers by using the "circle" emote.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/emote.dm
Comment (lzimann) [inline code/modules/mob/living/emote.dm]: add an `if(.)` before these new things

**Comment by GunHog** [comment]
Seems pointless and weird. I support.

**Review by Cyberboss — DISMISSED** [review]

**Comment by Niknakflak** [comment]
The game is much better than the circle.

Lame

**Review by Cyberboss — DISMISSED** [review]

**Comment by dannno** [comment]
This is a wrinklebrain PR.

**Comment by PKPenguin321** [comment]
I'm working on some tweaks to this, don't merge

**Comment by PKPenguin321** [comment]
Added a new thing that allows items to have unique actions reactions when they're equipped on a mob and the mob is examined. Also allows certain items to have unique examine text when on mobs. Circle game uses this to tell you when people examine you while you're making a circle.

File: code/modules/mob/living/emote.dm
Comment (PKPenguin321) [inline code/modules/mob/living/emote.dm]: This broke it and made it never work properly, so I removed it

File: code/game/objects/items/weapons/weaponry.dm
Comment (PKPenguin321) [inline code/game/objects/items/weapons/weaponry.dm]: I don't think this parent call works and I'm not sure why, help would be appreciated. Basically the special_examine_text doesn't show properly for the circled hand in particular, but the special_examine_text system works fine for everything else.

File: code/modules/mob/living/emote.dm
Comment (PKPenguin321) [inline code/modules/mob/living/emote.dm]: Talked to leoz on IRC and the if(.) is not actually needed so please disregard this review

File: code/game/objects/items.dm
Comment (Cyberboss) [inline code/game/objects/items.dm]: Remove the ` = null`

File: code/game/objects/items.dm
Comment (Cyberboss) [inline code/game/objects/items.dm]: TRUE

File: code/game/objects/items.dm
Comment (Cyberboss) [inline code/game/objects/items.dm]: FALSE

File: code/game/objects/items/weapons/weaponry.dm
Comment (Cyberboss) [inline code/game/objects/items/weapons/weaponry.dm]: Should just be `return ..()` I think

File: code/game/objects/items/weapons/weaponry.dm
Comment (Cyberboss) [inline code/game/objects/items/weapons/weaponry.dm]: I hope you realize these to_chats happen before any of the default examine text

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/weaponry.dm
Comment (PKPenguin321) [inline code/game/objects/items/weapons/weaponry.dm]: Yeah. Not sure how to fix

Comment (octareenroon91) [inline code/game/objects/items/weapons/weaponry.dm]: You could devolve more of the examine logic into `examine_reaction()`.
`/mob/living/carbon/human/examine(mob/user)`uses `to_chat()` and doesn't return, so you could break the output into multiple to_chat instances.

Comment (octareenroon91) [inline code/game/objects/items/weapons/weaponry.dm]: Wait, I'm wrong. 
1. the userdanger line goes into the examine msg stream. 
2. the to-owner message goes now.

**Comment by PKPenguin321** [comment]
Undid shitty implementation of examining stuff. Maybe some other time.

Implementation is now clean and this should be ready to go.

**Comment by Xhuis** [comment]
I don't understand the hate going around about this. It's a harmless, fun little feature, and the framework it adds can be useful for other stuff, too. We have maintainers here _refusing_ to help, and not answering when PK offered to just add the framework.

Why get so upset over something so small?

**Comment by demonfiren** [comment]
well remie hates it due to getting bopped as per forum post

**Comment by dannno** [comment]
I'm disappointed in the behaviour of our maintainers towards Pkp in regards to this situation.
Come on.

**Comment by Xhuis** [comment]
Cyberboss has been helpful in keeping the code modern and up-to-date, and I think he shouldn't be complained about here. Others are refusing to help due to a personal bias, which seems wrong to me, but then again, I'm not a maintainer. I do think that the maintainers who dislike the feature should say why and give good reasons other than their own personal dislike (while they're entitled to that, one maintainers' dislike is no reason to close a PR on subject alone.)

**Comment by Cyberboss** [comment]
I have nothing against this PR, I'll merge it if a good solution is presented. I just don't really have the spare energy to come up with it myself

**Comment by PKPenguin321** [comment]
I solved it by removing the examine functionality since it was never required in the first place

**Review by Cyberboss — APPROVED** [review]

**Comment by Xhuis** [comment]
Hmmm. I would add that framework back in, but it already caused a lot of drama alone and there doesn't seem to be an easy way for it to be implemented.

@PKPenguin321 Don't, like, flaunt the fact that this got merged. I know of at least two maintainers that this is going to make upset on the mere content alone; no reason to stoke the fires more than they have to be. I do hope that they see that it's just a harmless thing that has really zero impact and is just something for people to have fun with, as that might sway them.

**Comment by demonfiren** [comment]
Fun is illegal, though.

**Pull request merged**
